src/ziggurat/ssl/properties.clj

```diff
@@ -0,0 +1,44 @@
+(ns ziggurat.ssl.properties
+  (:require [ziggurat.config :refer [ziggurat-config]]
+            [ziggurat.ssl.constants :as const])
+  (:import [org.apache.kafka.streams StreamsConfig]
+           [org.apache.kafka.common.config SslConfigs SaslConfigs]))
+
+(defn- ssl-configs
+  [key]
+  (get-in (ziggurat-config) [:ssl-config key]))
+
+(defn- validate-mechanism
+  [mechanism]
+  (when-not (contains? #{const/mechanism-plain const/mechanism-scram-512} mechanism)
+    (let [message (format "SSL mechanism can either be %s or %s" const/mechanism-plain const/mechanism-scram-512)]
+      (throw (IllegalArgumentException. message {:mechanism mechanism})))))
+
+(defn- validate-protocol
+  [protocol]
+  (when-not (contains? #{const/protocol-plaintext const/protocol_sasl_ssl} protocol)
+    (let [message (format "protocol can either be %s or %s" const/protocol-plaintext const/protocol_sasl_ssl)]
+      (throw (ex-info message {:protocol protocol})))))
+
+(defn- create-jaas-properties
+  [user-name password mechanism]
+  (validate-mechanism mechanism)
+  (let [jaas-template (const/get-jaas-template mechanism)]
+    (format "%s required username=\"%s\" password=\"%s\";" jaas-template user-name password)))
+
+(defn build-ssl-properties
+  [properties]
+  (let [enabled (ssl-configs :enabled)
```

Comment:
Ziggurat follows a convention for defining keys for kafka properties i.e. original kafka properties are converted to  small case, underscores are replace with hyphens and the resulting string is converted to a keyword.

Please check the code in config.clj to understand how this convention enables Ziggurat to pick up just any kafka property. Specifically, the PR should use `set-property` method in config.clj to parse the keywords as kafka configs.